lib/zone/zone.ts

```diff
@@ -88,17 +88,64 @@ export interface Zone {
+    /// </remarks>
     broadcast(func: (...args: any[]) => void, args?: any[]) : Promise<void>;
 
+    /// <summary> Run the function on all workers with the given arguments. </summary>
+    /// <param name="func"> The JS function. </param>
+    /// <param name="args"> The arguments that will pass to the function. </param>
+    /// <returns> A promise which is resolved when broadcast completes and rejected when failed. </returns>
+    /// <remarks>
+    ///     Broadcast is designed for the purpose of bootstrapping/changing internal state on all workers.
+    ///     Function broadcast returns a promise of void, telling whether the operation succeeded or failed.
+    ///     Promise will be rejected on failure from any worker, though most likely all workers will fail if one fails.
+    /// </remarks>
+    broadcast(func: (...args: any[]) => Promise<void>, args?: any[]) : Promise<void>;
+
+    /// <summary> Compiles and run the provided source code on all zone workers synchronously. </summary>
+    /// <param name="source"> A valid javascript source code. </param>
+    /// <remarks>
+    ///     Broadcast is designed for the purpose of bootstrapping/changing internal state on all workers.
+    ///     Function broadcastSync wait for all workers to complete.
+    ///     An exception will be thrown when any worker fails.
+    /// </remarks>
+    broadcastSync(source: string) : void;
+
+    /// <summary> Run the function on all workers with the given arguments. </summary>
+    /// <param name="func"> The JS function. </param>
+    /// <param name="args"> The arguments that will pass to the function. </param>
+    /// <returns> A promise which is resolved when broadcast completes and rejected when failed. </returns>
+    /// <remarks>
+    ///     Broadcast is designed for the purpose of bootstrapping/changing internal state on all workers.
+    ///     Function broadcast returns a promise of void, telling whether the operation succeeded or failed.
+    ///     Promise will be rejected on failure from any worker, though most likely all workers will fail if one fails.
+    /// </remarks>
+    broadcastSync(func: (...args: any[]) => void, args?: any[]) : void;
+
+    /// <summary> Run the function on all workers with the given arguments. </summary>
+    /// <param name="func"> The JS function. </param>
+    /// <param name="args"> The arguments that will pass to the function. </param>
+    /// <returns> A promise which is resolved when broadcast completes and rejected when failed. </returns>
+    /// <remarks>
+    ///     Broadcast is designed for the purpose of bootstrapping/changing internal state on all workers.
+    ///     Function broadcast returns a promise of void, telling whether the operation succeeded or failed.
+    ///     Promise will be rejected on failure from any worker, though most likely all workers will fail if one fails.
+    /// </remarks>
+    broadcastSync(func: (...args: any[]) => Promise<void>, args?: any[]) : void;
+
```

Comment:
can we also provide broadcast(module: string, func, string, args?:any[], options?:CallOptions) : Promise<Result>
and  broadcast(module: string, func, string, args?:any[], options?:CallOptions) ?

Reply:
Not going to support this, for 2 reasons:
1- broadcast does not support transport context. This usage is highly depending on argument passing, which does not work well in broadcast
2- eventually this type of API will be removed. passing a function may be the best option since it is straightforward, flexible and more javascript-style. and it will cover all the ability of current APIs as we have the module system ready.